drivers/spi/spi_gecko_eusart.c

```diff
@@ -0,0 +1,325 @@
+			       : ((ref) == EUSART1) ? cmuClock_EUSART1 \
+			       : ((ref) == EUSART2) ? cmuClock_EUSART2 \
+			       : -1)
+#elif (EUSART_COUNT == 4)
+#define CLOCK_EUSART(ref)	(((ref) == EUSART0) ? cmuClock_EUSART0 \
+			       : ((ref) == EUSART1) ? cmuClock_EUSART1 \
+			       : ((ref) == EUSART2) ? cmuClock_EUSART2 \
+			       : ((ref) == EUSART3) ? cmuClock_EUSART3 \
+			       : -1)
+#else
+#error "Undefined number of EUSARTs."
+#endif /* EUSART_COUNT */
+#define GET_GECKO_EUSART_CLOCK(id) CLOCK_EUSART((EUSART_TypeDef *)DT_INST_REG_ADDR(id))
+#endif /* DT_NODE_HAS_PROP(n, peripheral_id) */
+
+#define SPI_WORD_SIZE 8
+
+/* Structure Declarations */
+
+struct spi_gecko_data {
+	struct spi_context ctx;
+};
+
+struct spi_gecko_config {
+	EUSART_TypeDef *base;
+	CMU_Clock_TypeDef clock;
+	const struct pinctrl_dev_config *pcfg;
+};
+
+
+/* Helper Functions */
+static int spi_config(const struct device *dev,
+		      const struct spi_config *config,
+		      uint16_t *control)
+{
+	struct spi_gecko_data *data = dev->data;
+	const struct spi_gecko_config *gecko_config = dev->config;
+
+	EUSART_SpiAdvancedInit_TypeDef eusartAdvancedSpiInit = EUSART_SPI_ADVANCED_INIT_DEFAULT;
+	EUSART_SpiInit_TypeDef eusartInit = EUSART_SPI_MASTER_INIT_DEFAULT_HF;
+
+	if (spi_context_configured(&data->ctx, config)) {
+		/* Already configured. No need to do it again. */
+		return 0;
+	}
+
+	if (config->operation & SPI_HALF_DUPLEX) {
+		LOG_ERR("Half-duplex not supported");
+		return -ENOTSUP;
+	}
+
+	if (SPI_WORD_SIZE_GET(config->operation) != SPI_WORD_SIZE) {
+		LOG_ERR("Word size must be %d", SPI_WORD_SIZE);
+		return -ENOTSUP;
+	}
+
+	if (IS_ENABLED(CONFIG_SPI_EXTENDED_MODES) &&
+	    (config->operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
+		LOG_ERR("Only supports single mode");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_TRANSFER_LSB) {
+		LOG_ERR("LSB first not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_OP_MODE_SLAVE) {
+		LOG_ERR("Slave mode not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_MODE_LOOP) {
+		eusartInit.loopbackEnable = eusartLoopbackEnable;
+	} else {
+		eusartInit.loopbackEnable = eusartLoopbackDisable;
+	}
+
+	/* Set Clock Mode */
+	if (config->operation & SPI_MODE_CPOL) {
+		if (config->operation & SPI_MODE_CPHA) {
+			eusartInit.clockMode = eusartClockMode3;
+		} else {
+			eusartInit.clockMode = eusartClockMode2;
+		}
+	} else {
+		if (config->operation & SPI_MODE_CPHA) {
+			eusartInit.clockMode = eusartClockMode1;
+		} else {
+			eusartInit.clockMode = eusartClockMode0;
+		}
+	}
+
+	if (config->operation & SPI_CS_ACTIVE_HIGH) {
+		eusartAdvancedSpiInit.csPolarity = eusartCsActiveHigh;
+	} else {
+		eusartAdvancedSpiInit.csPolarity = eusartCsActiveLow;
+	}
+
+	eusartInit.databits = eusartDataBits8;
+	eusartInit.advancedSettings = &eusartAdvancedSpiInit;
+
+	/* Configure the clocks. */
+	CMU_ClockSelectSet(cmuClock_EM01GRPCCLK, cmuSelect_HFRCODPLL);
```

Comment:
While EUSART1 uses the Group C clock on xG24, EUSART0 is configurable. But it defaults to the Group C clock, so it's technically workable to not expose that configurability in Zephyr.

However, other devices have the EUSART at different nodes in the clock tree: for instance, xG27 uses the Group A clock, and doesn't have a Group C clock.

The Group A clock is currently configured in `board.c`. By similarity, until a proper clock control driver is created for EFR32, the Group C clock should also be configured there for the 3 xG24 based boards.

Reply:
Done, moved the config to board.c

Thanks!